File: main-modules/common/src/main/java/com/sidework/common/util/CursorUtil.java
package com.sidework.common.util;

import java.time.Instant;
import java.util.Base64;

public class CursorUtil {

    public static String encode(CursorWrapper cursor) {
        Instant createdAt = cursor.cursorCreatedAt();
        Long id = cursor.cursorId();

        if (createdAt == null || id == null) return null;

        String raw = createdAt.toEpochMilli() + ":" + id;
        return Base64.getUrlEncoder().withoutPadding()
                .encodeToString(raw.getBytes());
    }

    public static CursorWrapper decode(String cursor) {
        if (cursor == null || cursor.isBlank()) {
            return new CursorWrapper(null, null);
        }

        try {
            String raw = new String(Base64.getUrlDecoder().decode(cursor));
            String[] parts = raw.split(":");
            Instant instant = Instant.ofEpochMilli(Long.parseLong(parts[0]));
            Long id = Long.parseLong(parts[1]);
            return new CursorWrapper(instant, id);
        } catch (Exception e) {
            return new CursorWrapper(null, null);
        }
    }
}

File: main-modules/common/src/main/java/com/sidework/common/util/CursorUtil.kt
//package com.moyeobus.global.util
//
//import java.time.Instant
//import java.util.Base64
//
//data class CursorWrapper(
//    val cursorCreatedAt: Instant?,
//    val cursorId: Long?
//)
//
//object CursorUtil {
//    fun encode(cursor: CursorWrapper): String? {
//        val (createdAt, id) = cursor
//        if (createdAt == null || id == null) return null
//        val raw = "${createdAt.toEpochMilli()}:$id"
//        return Base64.getUrlEncoder().withoutPadding().encodeToString(raw.toByteArray())
//    }
//
//    fun decode(cursor: String?): CursorWrapper {
//        if (cursor.isNullOrBlank()) return CursorWrapper(null, null)
//
//        return try {
//            val raw = String(Base64.getUrlDecoder().decode(cursor))
//            val (ts, id) = raw.split(":")
//            CursorWrapper(
//                Instant.ofEpochMilli(ts.toLong()),
//                id.toLong()
//            )
//        } catch (_: Exception) {
//            CursorWrapper(null, null)
//        }
//    }
//}